### DISCUSSION TITLE
更新日志 - 22.03.7 - 20240802

### DISCUSSION THREAD

**Discussion by jjm2473** [istoreos/istoreos#1737] [Announcements]
# 上游
**同步 OpenWRT [22.03.7](https://github.com/openwrt/openwrt/releases/tag/v22.03.7) 更新**。
小版本升级，一般不会有什么兼容性问题，要是万一遇到某些自己之前安装的程序无法运行，可以试试卸载重装相应插件。

# 设备
* 树莓派4：修复 /boot/config.txt 自定义配置在升级时会被重置的问题 #1721

**Comment by haobinnan** [comment]
固件包官方网站还没上传。

**Reply by jjm2473** [reply]
官网固件是其他人负责上传的，有时候忘了。可以先从这里下载最新固件 https://sourceforge.net/projects/istoreos/files/iStoreOS/ ，但是升级以后暂时别用在线升级，或者注意看版本号，以免又降级到老版本。

**Comment by bg4evk** [comment]
啥时候可以更新23.05

**Reply by jjm2473** [reply]
打算等24版本出来再升级

**Reply by haobinnan** [reply]
> 打算等24版本出来再升级

期待中。

**Reply by dunkeng** [reply]
> 打算等24版本出来再升级

https://openwrt.org/releases/22.03/notes-22.03.7
但openwrt官方说22.03.07后就不更新了，就算有BUG，然后我没找到他们什么时候更新24版的消息
OpenWrt 22.03 is EOL
:!: The OpenWrt 22.03 series is end of life according to the [OpenWrt security policy](https://openwrt.org/docs/guide-developer/security#support_status). The last release from the OpenWrt 22.03 series is 22.03.7, after this date we will not provide any updates for OpenWrt 22.03, not even for severe security problems. We encourage everyone to upgrade to OpenWrt 23.05 which will be supported till 2025.

**Reply by jjm2473** [reply]
@dunkeng istoreos 22.03又不是不维护了。
openwrt 23.05使用5.15内核，并没有太大变化，如果要支持还要重新适配rk35xx的内核，还不如等下一个版本直接上6.x。

**Reply by zzzbaba** [reply]
支持直接上6.X，是好事
但我作为旁路由使用来说，用了其他几个6.X的版本，总是存在防火墙相关的问题，表现为测速只有IPV6或者把梯子IP当作WAN IP的奇怪现象，我判断适配过程可能需要时间，尤其是23.05的防火墙替换掉了IPTABLE，请作者关注旁路由的适配问题

**Reply by JwietPuj-Drin** [reply]
> 支持直接上6.X，是好事 但我作为旁路由使用来说，用了其他几个6.X的版本，总是存在防火墙相关的问题，表现为测速只有IPV6或者把梯子IP当作WAN IP的奇怪现象，我判断适配过程可能需要时间，尤其是23.05的防火墙替换掉了IPTABLE，请作者关注旁路由的适配问题

！！！“把梯子IP当作WAN IP”这原来是防火墙的问题吗？我遇到这情况好久了，看 openclash 那边的 issue 区也看不出什么端倪

**Comment by cary-jjw** [comment]
还有一个硬盘挂载的问题 ，istoreos 磁盘挂载 重启后无法显示挂载硬盘 一定要格式化后才能再次加载！

**Comment by molinghu** [comment]
下个更新应该就是跟着上游到23.05了吧^v^

**Comment by javycoder** [comment]
这个版本升级总会下载失败，有没有别的办法升级[
![微信截图_20240807120413](https://github.com/user-attachments/assets/37161520-2e90-4e10-9d2d-b83be4a1cbe7)
](url)

**Reply by jjm2473** [reply]
这里可以手动下载 https://sourceforge.net/projects/istoreos/files/iStoreOS/

**Reply by shuowong1981** [reply]
你好，我用这个版本跑网速跑不满，千兆的网络实际跑700左右，我x86，n5000es，四个i226，电脑直连光猫可以1200，光猫是2.5g

**Reply by jjm2473** [reply]
x86可以试试在“网络”-“接口”-“全局设置”里面勾选数据包引导，然后重启下

**Comment by fbigun** [comment]
怎么手动升级，web端一直不能完成升级。可以下载完成并点击刷写固件，等重新重启到登录界面还是发现未升级。

**Reply by jjm2473** [reply]
看看有没有这个文件 /overlay/upgrade.log ，有的话提取出来或者粘贴下内容。

**Reply by fbigun** [reply]
```
cat /overlay/upgrade.log
2024-05-12 15:03:24
Running Version: 'iStoreOS 22.03.6 2024030112'
Sending TERM to remaining processes ...
Sending signal TERM to S99adb-enablemo (15044)
Sending signal TERM to adb (15054)
Sending signal TERM to adb (15064)
Sending signal TERM to dockerd (15285)
Sending signal TERM to pushbot (16071)
Sending signal TERM to sleep (17883)
Sending signal TERM to logread (18030)
Sending signal TERM to sleep (18208)
Sending signal TERM to zerotier-one (32068)
Sending KILL to remaining processes ...
Sending signal KILL to logread (18030)
Switching to ramdisk...
Performing system upgrade...
Extract boot sector from the image
Reading partition table from image...
Writing image to /dev/sda1...
Writing image to /dev/sda2...
Skip partition 3 >= 3 when upgrading
Writing new UUID to /dev/sda...
Upgrading bootloader on /dev/sda...
Upgrade completed
```

没有我刷写的日志，不知道怎么手工刷写，保留配置的那种

**Reply by jjm2473** [reply]
@fbigun 上一次日志还是05-12的，没有日志的话，可能你的系统本身就不正常。你是不是自己安装了其他源的base-files？参考 https://github.com/istoreos/istoreos/wiki/FAQ 第2点修复系统组件以后再试试。

**Reply by fbigun** [reply]
安装工具后，选择修复后出现异常

```
404 未找到
对不起，请求的目标未找到。

No page is registered at '/admin/system/systools/pages'. If this url belongs to an extension, make sure it is properly installed. If the extension was recently installed, try removing the /tmp/luci-indexcache file.

无法调度: /cgi-bin/luci/admin/system/systools/pages
```

**Reply by fbigun** [reply]
已安装的 systools 工具在重启后，消失

**Reply by jjm2473** [reply]
重刷吧

**Reply by fbigun** [reply]
不知道咋玩坏了的

**Reply by fbigun** [reply]
怎么在终端下直接刷机啊，在终端下直接执行，可否。  zcat istoreos.img.gz | dd of=/dev/sda bs=1MB

**Reply by jjm2473** [reply]
`zcat istoreos.img.gz | dd of=/dev/sda bs=1MB`理论上可行，但是理论上你没把根目录卸载掉就写入，数据有可能被破坏。
可以试试把固件放在/tmp/istoreos.img.gz，然后执行`sysupgrade -n -p -F /tmp/istoreos.img.gz`

**Reply by JwietPuj-Drin** [reply]
> 怎么在终端下直接刷机啊，在终端下直接执行，可否。 zcat istoreos.img.gz | dd of=/dev/sda bs=1MB

我向来就是这么刷的，还以为是标准操作呢……

**Comment by shuowong1981** [comment]
就是他的问题发自我的 iPhone在 2024年8月21日，15:59，Liangbin Lian ***@***.***> 写道：﻿
x86可以试试在“网络”-“接口”-“全局设置”里面勾选数据包引导，然后重启下

**Comment by bh10086** [comment]
下次能加个oscam读卡程序上去吗

**Reply by jjm2473** [reply]
应该可以自己用opkg或者软件包界面安装。这种程序实在是太小众了，不会加到istore的。

**Reply by jjm2473** [reply]
软件包那边好像没有，不过有docker镜像 https://docs.linuxserver.io/images/docker-oscam/

**Reply by jjm2473** [reply]
我估计没几个人玩卫星电视的，oscam这种高级货就更不用说了。

**Comment by rayangl** [comment]
有计划更新一下Docker吗

**Reply by jjm2473** [reply]
没，一般跟着openwrt上游更新版本。现在有哪个镜像需要依赖docker的版本吗？

**Reply by rayangl** [reply]
这倒是没有，是看到新版修复了很多漏洞和功能更新，所以问问咱们这边有没有跟进的计划。

**Reply by JwietPuj-Drin** [reply]
> 没，一般跟着openwrt上游更新版本。现在有哪个镜像需要依赖docker的版本吗？

@jjm2473 我知道 immich 的示例 compose 有个 healthcheck.start_interval 需要至少 Docker 25

**Comment by 357153645** [comment]
be200 网卡能驱动吗？ 我插上去都识别不出来

**Reply by jjm2473** [reply]
插上去执行`lspci -nnk`命令看看输出什么

**Reply by 357153645** [reply]
```
root@iStoreOS:~# lspci -nnk
0000:00:00.0 Host bridge [0600]: Intel Corporation 440BX/ZX/DX - 82443BX/ZX/DX Host bridge (AGP disabled) [8086:7192] (rev 03)
lspci: Unable to load libkmod resources: error -12
0000:00:07.0 ISA bridge [0601]: Intel Corporation 82371AB/EB/MB PIIX4 ISA [8086:7110] (rev 01)
        Subsystem: Microsoft Corporation Device [1414:0000]
0000:00:07.1 IDE interface [0101]: Intel Corporation 82371AB/EB/MB PIIX4 IDE [8086:7111] (rev 01)
        Kernel driver in use: ata_piix
0000:00:07.3 Bridge [0680]: Intel Corporation 82371AB/EB/MB PIIX4 ACPI [8086:7113] (rev 02)
0000:00:08.0 VGA compatible controller [0300]: Microsoft Corporation Hyper-V virtual VGA [1414:5353]
        Kernel driver in use: hyperv_fb
acf0:00:00.0 Network controller [0280]: Intel Corporation Wi-Fi 7(802.11be) AX1775*/AX1790*/BE20*/BE401/BE1750* 2x2 [8086:272b] (rev 1a)
        Subsystem: Intel Corporation BE200 320MHz [Gale Peak] [8086:00f4]
root@iStoreOS:~# 
```

**Reply by jjm2473** [reply]
看起来没驱动成功，开机以后尽快执行`dmesg | grep iwlwifi`看看结果。不过Intel的wifi网卡都是鸡肋，没法作为5G AP使用，如果你想用这个作为AP可以放弃了。

**Reply by 357153645** [reply]
root@iStoreOS:~# dmesg | grep iwlwifi
[   11.892072] auto rmmod iwlwifi

**Reply by 357153645** [reply]
驱动有 但是就是驱动不起来  自动卸载了驱动   我知道不能做ap  应该要可以驱动起来呀

**Reply by 357153645** [reply]
我尝试过openwrt  可以驱动  但是工作起来也还是有问题 会自动断线。

**Comment by rayangl** [comment]
![image](https://github.com/user-attachments/assets/ab662897-1e28-4b08-925e-2e1ec2fee5af)
能否优化一下移动端的编辑体验，手机浏览器UA不论是手机模式还是电脑模式，不能正常编辑。

1. 插入光标以后，页面被放大，并且无法缩小和滑动。
2. 保存按钮太靠下，基本点不到，容易点到编辑框，页面又会被放大。

**Comment by jaylove520** [comment]
我想增加一个检测ipv6是否连通的，如果不通重新拨号！因为经常性会只有ipv4地址，而ipv6被运营商踢下线了，然后导致全部设备无法上网，只能重新拨号解决！基本所有openwrt都有这个问题，现在我是ifup wan来解决，能不能做成功能啊！期待6.x内核版本

**Comment by zjzj52** [comment]
友善的r2s频繁掉线，主要表现就是LAN口指示灯不闪了，数据流量稍微大一点出现此问题的概率就高。基本上一天要掉一次，时间长的时候差不多3天掉一次。

**Reply by hzhanghust** [reply]
我也是 r2s 基本上平均 2 天左右就出现一次一样的现象

**Comment by jaylove520** [comment]
您好，我已经接收到您的邮件，我将尽快查阅并给您回复。

**Comment by pilgrimage233** [comment]
你好，请问可以增加shortcut-FE加速组件吗，现在只有Linux的flow offloading，希望可以给用户更多选择！

**Reply by jjm2473** [reply]
shortcut-FE早就被弃用了，4.x时代的东西

**Comment by Guoziguoran** [comment]
请问一下：最近IP地址老是不定时变动，日志显示eth0链路不稳定或者故障，到底是我系统问题还是网络自身的问题？
![IMG_2257](https://github.com/user-attachments/assets/c0c5b81c-77b5-420e-92b8-42ecdd9e5756)

**Comment by molinghu** [comment]
OpenWRT 24.10正式版发布了，请问什么时候出第一个基于此的istoreos？

**Comment by jaylove520** [comment]
您好，我已经接收到您的邮件，我将尽快查阅并给您回复。
